### ISSUE TITLE
Very large hints returned sometimes

### ISSUE THREAD

**Issue by alanz** [ndmitchell/hlint#375]
If a large function is defined something like

```haskell
foo x = do
  case x of
     -- many many lines of case matching and processing, where clauses, etc
```
then the `redundant do` hint will repeat the entire text for function `foo` twice, once with the `do` and once without.

This can result in unwieldy hints, especially when they are presented as a pop-up on hovering in an IDE.

**Comment by ndmitchell** [comment]
I don't disagree the hints are big - but I don't see a "good" way to trim them generically. Perhaps the IDE should be doing that, as it "knows" what big is from its context?

**Comment by alanz** [comment]
That is an option. I think `apply-refact` may also use the full text.  Realistically you only need the first line after the `do`, for this particular hint, to get the message across.

Perhaps at some future date the hints could have some sort of output control, to state how much context should be provided.

I do not know the details of the inside of hlint, I do know that it is pretty complex. And I do not have much expectation of this changing, just wanted to report a niggle.

**Comment by ndmitchell** [comment]
`apply-refact` uses the JSON output, so I don't think that's too much of a problem. I do truncate some hints,  e.g.

    yes_foo = code goes here

Suggests

    yesFoo = ...

Maybe this do is just another example. However, I think in the limit almost every hint can be arbitrarily long.
